docs/source/development.rst

```diff
@@ -106,14 +106,164 @@ are making:
 
     $ git remote add upstream https://github.com/ga4gh/server.git
 
+All development is done against the ``develop`` branch.  The ``master``
+branch is meant to be kept stable since it is the branch releases are
+based on -- don't touch it!  These are the two mainline branches.
+Our branching model is loosely based on the one described
+`<here http://nvie.com/posts/a-successful-git-branching-model/>`_.
+
+All development should be done in a topic branch.  That is, a branch
+that the developer creates him or herself.  These steps will create
+a topic branch (assuming that ``upstream`` points to 
+``https://github.com/ga4gh/server.git``) and ``origin`` points to 
```

Comment:
Is this parenthetical comment a bit confusing? We've just told the user to create a remote called upstream in the previous paragraph, and I think we can assume origin follows the usual convention.